### PR TITLE
[RP] Chapter 04 Scala Python Ported code.

### PR DESCRIPTION
**Pull request by rambabu-posa**

### CODE REVIEW

File: src/main/scala/net/jgp/books/spark/ch04/lab900_misc/Hello_WorldApp.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package net.jgp.books.spark.ch04.lab900_misc
+
+/**
+  * Hello world.
+  *
+  * @author rambabu.posa
+  */
+object Hello_WorldApp {
```
Comment (jgperrin) [inline src/main/scala/net/jgp/books/spark/ch04/lab900_misc/Hello_WorldApp.scala]: Is that an acceptable name in Scala? It is not in Java (_)

File: README.md [hunk 2]
```diff
@@ -6,7 +6,34 @@ Welcome to Spark with Java, chapter 4. This chapter is about laziness.
...
+
+    1. Clone this project
+       Assume that cloned this project to ${MY_HOME_DIR}
+
+    2. cd ${MY_HOME_DIR}
+
+    3. Create application jar file
+       ```mvn clean package```
+
+    3. Execute the following spark-submit command to create a jar file to our this application
```
Comment (jgperrin) [inline README.md]: 4?

File: pom.xml [hunk 3]
```diff
@@ -12,12 +12,19 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <java.version>1.8</java.version>
     <scala.version>2.11</scala.version>
+    <scala.full.version>2.11.11</scala.full.version>
```
Comment (jgperrin) [inline pom.xml]: What about we have a pom-scala.xml for the Scala code? It will be a lot clearer for the reader.

**Review by jgperrin — CHANGES_REQUESTED** [review]
Can you get rid of the __init__.py? Maybe "cd" to where the file is and run it from there?

File: src/main/scala/net/jgp/books/spark/ch04/lab900_misc/Hello_WorldApp.scala (hunk 1 above)
Comment (rambabu-posa) [reply src/main/scala/net/jgp/books/spark/ch04/lab900_misc/Hello_WorldApp.scala]: Yes It works in Scala. As we need to differentiate Java and Scala applications, we need to use different package name or class name.

File: README.md (hunk 2 above)
Comment (rambabu-posa) [reply README.md]: Yes typo

File: pom.xml (hunk 3 above)
Comment (rambabu-posa) [reply pom.xml]: Apologies, never used such kind of pom-scala.xml. Here we've just added only two new entries: one dependency and one plugin. Please suggest me on this

**Pull request merged**